src/NuGet.Core/NuGet.Protocol/Resources/RawSearchResourceV3.cs

```diff
@@ -103,22 +108,84 @@ public virtual async Task<JObject> SearchPage(string searchTerm, SearchFilter fi
                     throw new FatalProtocolException(string.Format(CultureInfo.CurrentCulture, Strings.Protocol_BadSource, queryUrl.Uri), ex);
                 }
 
-                if (searchJson != null)
+                if (searchResult != null)
                 {
-                    return searchJson;
+                    return searchResult;
                 }
             }
 
             // TODO: get a better message for this
             throw new FatalProtocolException(Strings.Protocol_MissingSearchService);
         }
 
-        public virtual async Task<IEnumerable<JObject>> Search(string searchTerm, SearchFilter filters, int skip, int take, Common.ILogger log, CancellationToken cancellationToken)
```

Comment:
Here's another public API that was removed, and should not be.

Reply:
@zivkan would you like me put it back along new API?

Reply:
yes, please try to make sure any public method/constructor/property/field in any public class in `src\NuGet.Core\*` does not change, so any assemblies compiled against older versions of the DLL do not crash if this new DLL is dropped in place.

If it turns out to be very difficult, we'll reconsider, but always try to maintain API & ABI compatibility, and try to avoid adding new APIs, where possible, as well. Working on something that is both a product and an SDK can be frustrating :(

Reply:
@nkolev92 Please give your input.  Do we can either have them side by side or remove the old one?

Reply:
I just realized that the old API is returning `JObject`, so I care a whole lot less.

However, I'm not sure that you modified anything that the old APIs used. So, I think if you simply copy the old methods and paste it back in this file, then it should be ok.  But mark it with the `Obsolete` attribute, and tell anyone using it to call `PackageSearchResource` instead.

If that doesn't work, I think you should be able to call the new API from the old API, then serialize the result into a string, and then deserialize it back into a JObject or JArray. Performance will suck, but again, we mark the methods as obsolete and tell them to use `PackageSearchResource` instead.  Unfortunately your new APIs have this new `Func` parameter, so it may be non-trivial to implement. But honestly, I find it confusing anyway, so if it was cleaned up, I think the code may be easier to understand anyway.

If it's all too hard, then I'm ok with breaking this API. I'm very unhappy with APIs using or return JSON LINQ.  I don't know if it's better to keep the old API and just throw NotImplementedException, or just delete the old API. Either way any binaries that already use it will fail.

Reply:
If we end up deleting the APIs, we should also make the class and constructor internal. A public class with no methods or properties is not very useful.  Although I'm hoping that copying the old methods will just work.

Reply:
@zivkan @nkolev92 I put back old API and marked as Obsolete so it wouldn't break other applications that depend on old json API.